aws/resource_aws_cloudwatch_log_stream.go

```diff
@@ -58,9 +58,17 @@ func resourceAwsCloudWatchLogStreamCreate(d *schema.ResourceData, meta interface
 func resourceAwsCloudWatchLogStreamRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).cloudwatchlogsconn
 
-	ls, exists, err := lookupCloudWatchLogStream(conn, d.Id(), d.Get("log_group_name").(string), nil)
+	group := d.Get("log_group_name").(string)
+
+	ls, exists, err := lookupCloudWatchLogStream(conn, d.Id(), group, nil)
 	if err != nil {
-		return err
+		_, err2 := lookupCloudWatchLogGroup(conn, group)
```

Comment:
Instead of performing another API call which should return the same result, can we just use the error itself to do the right thing? e.g.

```go
if isAWSErr(err, cloudwatchlogs.ErrCodeResourceNotFoundException, "") {
  exists = false
}
```

or within `lookupCloudWatchLogStream()`:

```go
	resp, err := conn.DescribeLogStreams(input)
	if isAWSErr(err, cloudwatchlogs.ErrCodeResourceNotFoundException, "") {
		return nil, false, nil
	}
	if err != nil {
		return nil, true, err
	}
```